build.gradle

```diff
@@ -44,14 +44,18 @@ dependencies {
 
     testImplementation gradleTestKit()
     testImplementation 'org.spockframework:spock-core:2.0-groovy-3.0'
-    testImplementation 'net.bytebuddy:byte-buddy:1.12.18'
+    testImplementation 'net.bytebuddy:byte-buddy:1.14.8'
     testImplementation 'org.objenesis:objenesis:3.3'
-    testImplementation 'commons-io:commons-io:2.11.0'
+    testImplementation 'commons-io:commons-io:2.14.0'
 
     functionalTestImplementation localGroovy()
     functionalTestImplementation gradleTestKit()
     functionalTestImplementation 'org.spockframework:spock-core:2.0-groovy-3.0'
-    functionalTestImplementation 'commons-io:commons-io:2.11.0'
+    functionalTestImplementation 'commons-io:commons-io:2.14.0'
+}
+
+java {
+    toolchain.languageVersion = JavaLanguageVersion.of(8)
```

Comment:
Let's make it 17 and I will release a major version.

Reply:
If we use 17 here, it requires all consumers running on Java 17 above, we should keep the compatibility of Java 8.

Reply:
Pin 8 here for compatibility, use Java 17 on CI at 8ff89d42fb64e50ff20231fc72ede3d19ee8bdc7.